torch/distributions/distribution.py

```diff
@@ -177,6 +182,9 @@ def _validate_log_prob_arg(self, value):
         if not (torch.is_tensor(value) or isinstance(value, Variable)):
             raise ValueError('The value argument to log_prob must be a Tensor or Variable instance.')
 
+        if not self.support.check(value):
```

Comment:
This is too expensive to check in general. I think we should save the `validate_args` arg to a member variable and only validate this if `.validate_args == True`. We might even gate all of `_validate_log_prob_arg()` behind `if self.validate_args`.